### PR TITLE
Make gitleaks scan work on windows

### PR DESCRIPTION
**Pull request by thewizzy**
Windows does not have stdout, which it was using.

Output to a file `gitleaks-results.json` in the scan dir. This made the most sense as it is removed after scanning any way and very very unlikely to have a name clash. If it does clash it is written after the scan is performed and will be overwritten with valid scan result data.

Scan results are cleaned after the scan is complete and results have been consumed.

Also added a few more defaults to make testing defaults across platforms easier.

### CODE REVIEW

**Comment by bplaxco** [comment]
@thewizzy looks good. The one thing I want to look at tweaking is dropping the results inside the repo. If someone was using it to scan a lot of local repos, I wouldn't want to dirty up their repos. I'm working on something [here](https://github.com/leaktk/scanner/pull/9) that should give you a space to put them. This morning I realized there's a bug on that branch for local repos, I'm going to try to fix that and merge it in Wed. After that you can use something like  `workspace.results_dir` as a tmp place to write files and they won't be in the repo. :+1:

**Comment by thewizzy** [comment]
@bplaxco Thanks for the feedback. I will wait until workspaces is merged and then rejig this. That is a very good idea, I had not thought of the local repos use case.

**Review by bplaxco — CHANGES_REQUESTED** [review]
It looks like there may be the one issue with the shasum string having trailing spaces, but other than that, I think it looks good ^_^

File: src/config.rs
Comment (thewizzy) [reply src/config.rs]: Removed the two trailing. Had been fiddling with 8.16.1 support and swapped it back before pushing!
Good eye!

**Review by bplaxco — APPROVED** [review]
Looks good to me ^_^

**Pull request merged**
